client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -608,4 +662,50 @@ private enum State
          */
         FINISHED,
     }
+
+    private static class HeartbeatingResultRows
+            implements ResultRows
+    {
+        private final Iterator<List<Object>> iterator;
+        private final boolean isNull;
+        private final Runnable heartbeat;
+
+        public HeartbeatingResultRows(ResultRows delegate, Runnable heartbeat)
```

Comment:
Can you add unit test for this one?

Reply:
I can try but I need to think how to do it. The default keep-alive interval is 30s

Reply:
Done.